### PR TITLE
Modify hdf5 simulations files. 

### PR DESCRIPTION
**Pull request by fschlueter**
The flavor contains now the type of th…e parent particle. The interaction type contains now, in case of a charge lepton, the type of energy loss.

Implement the recently contained changes to the hdf5 files: 

- [x] `flavors` contains now the PDG particle code of the parent (neutrinos or muon, tau leptons)
- [x] `interaction_type` not anymore contains the neutrino interaction type ("cc" or "nc") but can also contain the PDG code for the shower creating secondary (when a charge lepton produces a shower)
- [x] Store the parent energy (directly before energy loss) in `energies`

A few thoughts:
 - Do we want/need to bump the version number 
 - Is it a problem to store strings and int both in `interaction_type`

### CODE REVIEW

File: NuRadioMC/EvtGen/generator.py
Comment (shallmann) [inline NuRadioMC/EvtGen/generator.py:1374]: I don't think this line does what we want.
The `interaction_type` being `cc` or `nc` might play a role in what the Askaryan model makes out of it - but I haven't checked.


 Was unaware that you are taking over implementing these changes. @jannavischer in that case please coordinate and make sure you are not doing things twice.
 
  I think you want to get the flavor of the `parent` not the product. Might make sence to wait until the proposal 7 branch is merged - since the interface changes

**Review by shallmann — CHANGES_REQUESTED** [review]

File: NuRadioMC/EvtGen/generator.py
Comment (fschlueter) [reply NuRadioMC/EvtGen/generator.py:1374]: For the interaction type I thought we want the particle code of the produce (to know what initiated the shower). The flavour/code of  the parent is stored in the item "flavor"

@jannavischer, sorry I did not know that you were working on this. As part of writing a tool to convert nur to hdf5 files I thought starting with this actually. I am currently trying to understand how to get the energy of the muon and tau from proposal exactly before the energy loss

> The interaction_type being cc or nc might play a role in what the Askaryan model makes out of it - but I haven't checked.

Are you sure? Or do you confuse it with shower_type (which has to be em or had).

Comment (fschlueter) [reply NuRadioMC/EvtGen/generator.py:1374]: > Might make sence to wait until the proposal 7 branch is merged - since the interface changes

hm very good point.

**Comment by fschlueter** [comment]
Hello @shallmann and @jannavischer, I was a little bit too impatient and started with getting the lepton energy. I get something which seems somewhat reasonable (see attached plot) but not jet correct (other plots do not look so good). There are some subtleties when shower secondaries are merged... maybe we can discuss this on Monday?
lepton_energy_379

**Comment by fschlueter** [comment]
> Hello @shallmann and @jannavischer, I was a little bit too impatient and started with getting the lepton energy. I get something which seems somewhat reasonable (see attached plot) but not jet correct (other plots do not look so good). There are some subtleties when shower secondaries are merged... maybe we can discuss this on Monday?
> lepton_energy_379

For explanation, the blue curve shows what I am not getting from proposal as the energy of the lepton, the orange curve shows what one can calculate when one subtract the shower energies from the initial lepton energy (this ignores continues energy losses and/or energy losses below 1 PeV, hence it is expected that the two curves do not agree)

**Comment by fschlueter** [comment]
@cg-laser, with these changes we store str ("nc", "cc") and integer (pdg codes of shower primaries (e.g. 83 or 85) in the same item `interaction_type`. Do you think that is okay? Would you rather have that avoided?

**Comment by shallmann** [comment]
I would avoid having different types in one variable. Aren't we mixing `interaction-type identifiers` with `product codes` then? (Sorry I am slow here)

According to the existing documentation:
```
``interaction_type``, the interaction type. `'cc'` for charged current, and `'nc'` for neutral current. `'tau_had', 'tau_em', 'tau_mu'` indicate the tau decays into the hadronic, electromagnetic and muonic channels respectively, calculated using the CSDA for taus. If the vertex has been created by a secondary particle propagated by PROPOSAL, `had` indicates a hadronic shower, and `em` an electromagnetic shower.
```
it was always a string before; at least for the `tau_...` the only function that relies on that naming convention is `inelasticity_tau_decay(tau_energy, branch, rnd=None)` and that one is not used anywhere in the processing chain.

Not sure what is best to do.

**Comment by fschlueter** [comment]
> I would avoid having different types in one variable. Aren't we mixing `interaction-type identifiers` with `product codes` then? (Sorry I am slow here)
> 
> According to the existing documentation:
> 
> ```
> ``interaction_type``, the interaction type. `'cc'` for charged current, and `'nc'` for neutral current. `'tau_had', 'tau_em', 'tau_mu'` indicate the tau decays into the hadronic, electromagnetic and muonic channels respectively, calculated using the CSDA for taus. If the vertex has been created by a secondary particle propagated by PROPOSAL, `had` indicates a hadronic shower, and `em` an electromagnetic shower.
> ```
> 
> it was always a string before; at least for the `tau_...` the only function that relies on that naming convention is `inelasticity_tau_decay(tau_energy, branch, rnd=None)` and that one is not used anywhere in the processing chain.
> 
> Not sure what is best to do.

Okay it's exactly the last sentence in documentation we want to change. 
Last call, we agreed storing the particle code in the `interaction_type` which was previously stored (for secondary showers) in `flavor`. However, to be more consistent it makes sense to store the code of the parent particle (in this case the mu/tau lepton) in `flavor`. Hence, to keep all information we decided to store the code of the shower-initiating particle in `interaction_type`. Keep in mind that all other information are stored already. 

To "fix" the problem with the data type we could choose to not store the particle code (integer) but the particle names:
`particle_names.particle_name(product.code)`

**Comment by fschlueter** [comment]
This PR will be closed in the favour of #486 which does the same implementation directly for proposal7. 
This implementation is almost complete, just convert the `product.code` to a particle name to store it in `interaction_type`

**Pull request closed without merging**
